### ISSUE TITLE
The dialog.open method causes a memory leak

### ISSUE THREAD

**Issue by DominikPalo** [slackapi/node-slack-sdk#529]
### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

Every call of the method `dialog.open` causes a memory leak.

#### Reproducible in:

`@slack/client` version: 4.1.0

node version: 6.10.0

OS version(s): Ubuntu 16.04, Windows 10

#### Steps to reproduce:

1. Create a new instance of the WebClient
2. Call `client.dialog.open({...})`

#### Expected result:

A dialog is displayed.

#### Actual result:

A dialog is displayed, but memory usage of nodejs process starts increasing (about 30MB/sec) until the maximum JavasScript heap size is exceeded and then the process crashes.

#### Attachments:
```
[DEBUG] @slack/client:WebClient initialized
[DEBUG] @slack/client:WebClient apiCall() start
[DEBUG] @slack/client:WebClient request attempt
FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
```
Replicated on 3 different machines (local Windows, local Ubuntu and Ubuntu Server in AWS cloud)

**Comment by DominikPalo** [comment]
It seems that the same problem affects also other WebClient methods - e.g. `chat.postMessage`

**Comment by aoberoi** [comment]
Can you provide a script that exhibits this memory leak issue? I’ll do some memory profiling on the script to narrow down the cause of any leaks.

**Comment by DominikPalo** [comment]
After a few hours of investigating, debugging and trying to isolate a minimal example from our project I found that the problem was caused by the combination of HAPI.js and a specific version of the newrelic module. Memory leaks were happening only when a method of Slack WebClient was called in a HAPI.js route handler with `await`. But, after removing  the newrelic module library I wasn't able to replicate the problem. Then I updated the newrelic module from the version 2.4.0 to the version 2.9.1 and it helped - now all works fine, no memory leaks.
